### ISSUE TITLE
Genotype requires a method filtersWereApplied

### ISSUE THREAD

**Issue by heuermh** [samtools/htsjdk#741]
Where do the assertions in #663 come from?

"A Genotype always has filters. If there are no non-PASS filters, it's PASS."

Per VCF specification:
"FT : sample genotype filter indicating if this genotype was “called” (similar in concept to the FILTER field). Again, use PASS to indicate that all filters have been passed, a semi-colon separated list of codes for filters that fail, **or ‘.’ to indicate that filters have not been applied.** These values should be described in the meta- information in the same way as FILTERs (String, no white-space or semi-colons permitted)"

`Genotype` requires a method `boolean filtersWereApplied()` similar to that in `VariantContext` to support this distinction.

**Comment by fnothaft** [comment]
+1, subscribing

**Comment by yfarjoun** [comment]
I guess the assertion was a way to say that you should always be able to
ask about a genotype whether it is filtered or not.

do you think that there should also be a "wereFiltersApplied()" function
that would return false if the FT tag is missing or "."?

**Comment by heuermh** [comment]
> do you think that there should also be a "wereFiltersApplied()" function that would return false if the FT tag is missing or "."?

Yes, that would be my suggestion.  Because of the special treatment `FT` receives as one of `PRIMARY_KEYS` (https://github.com/samtools/htsjdk/blob/master/src/main/java/htsjdk/variant/variantcontext/Genotype.java#L54) the actual attribute value appears not to be accessible.

**Comment by heuermh** [comment]
This one has been sitting for a while, any update?

**Comment by yfarjoun** [comment]
I think that it's a reasonable idea. Genotype filters haven't loomed large in our world...(though this might soon change)...So I'm not sure that we would prioritize submitting the PR...but we will review it if you have a suggestion.
